doc/tools/format-tool.md

```diff
@@ -0,0 +1,125 @@
+# `FormatTool`
+
+Formats dart files in the current project by running `dartfmt`.
+
+## Usage
+
+> _This tool is included in the [`coreConfig`][core-config] and is runnable by
+> default via `ddev format`._
+
+```dart
+// tool/dart_dev/config.dart
+import 'package:dart_dev/dart_dev.dart';
+
+final config = {
+  'format': FormatTool() // configure as necessary
+};
+```
+
+## Default behavior
+
+By default this tool will run `dartfmt -w .` which will format all dart files in
+the current project.
+
+## Configuration
+
+### Default mode
+
+`FormatTool` can be run in 3 modes:
+
+1. `FormatMode.overwrite`(default)
+2. `FormatMode.dryRun` (lists files that would be changed)
+3. `FormatMode.check` (dry-run _and_ sets the exit code if changes are needed)
```

Comment:
Is it worth indicating here what the command line args are that represent these modes? Or just leave it up to the user to run `ddev help format` to find out?

Reply:
Yeah that wouldn't hurt, I'll add it